persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/RestoreBackupTest.kt

```diff
@@ -0,0 +1,487 @@
+    }
+
+    @Test
+    fun givenBackupHasAssetsButUserNot_whenRestoringBackup_thenBackupAssetsAreRestored() = runTest {
+        // given
+        val assetsToBackup = backupDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToBackup.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+    }
+
+    @Test
+    fun givenUserHasAssetButBackupNot_whenRestoringBackup_thenUserAssetAreRestored() = runTest {
+        // given
+        val assetsToRestore = userDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToRestore.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+    }
+
+    @Test
+    fun givenUserHasUniqueAssetsAlongWithBackup_whenRestoringBackup_thenThoseAssetsAreRestored() = runTest {
+        // given
+        val assetsToRestore = userDatabaseDataGenerator.generateAndInsertAssets(10)
+        val backupAssets = backupDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToRestore.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+        backupAssets.forEach { backupAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(backupAssetEntity.key).first()
+
+            assertTrue(asset != null)
+            assertEquals(asset, backupAssetEntity)
+        }
+    }
+
+    private suspend fun insertOverlappingConversations(amount: Int): List<ConversationEntity> {
+        val conversationAdded = mutableListOf<ConversationEntity>()
+
+        for (index in 1..amount) {
+            val overLappingId = ConversationIDEntity("overlappingValue$index", "overlappingDomain$index")
+            val overLappingName = "overLappingName$index"
+
+            val conversationType = ConversationEntity.Type.values()[index % ConversationEntity.Type.values().size]
+
+            val invalidatedConversationType =
+                if (conversationType == ConversationEntity.Type.CONNECTION_PENDING)
+                    ConversationEntity.Type.values()[(index + 1) % ConversationEntity.Type.values().size]
+                else conversationType
```

Comment:
Wouldn't this insert multiple `Conversation.Type. SELF` types of conversations instead of having just one?

Reply:
I am sorry but I do not follow the question

Reply:
This is basically looping around the conversation types, but substituting the `CONNECTION_PENDING` ones for the next type of conversation, i.e. `Conversation.Type.SELF`

Reply:
I am sorry but I still do not follow, yes this is what it is doing, making sure we do not use CONNECTION_PENDING because when we quuery all the conversations we will get less than we expect if we used CONNECTION_PENDING, therefore it would make the unit testsing not behave as expected

Reply:
I just wanted to randomize the data

Reply:
My point was that we should only have one `Conversation.Type.SELF` for each user account, not multiple ones. I'm not sure if this will break anything, but better be on the safe side and substitute `CONNECTION_PENDING` for `GROUP` or `ONE_ON_ONE` types.

Reply:
ahhhh I see, worth doing it I think

Reply:
good point, thanks